media/js/src/multipleChoiceQuestion2.jsx

```diff
@@ -53,12 +83,15 @@ export const MultipleChoiceQuestion2 = ({ isSubmitted, setIsSubmitted,
     useEffect(() => {
         if (isCorrect) {
```

Comment:
I would also take this save out of the useEffect. It can go on an onSubmit. Also we are saving answers regardless of whether the answer is correct. So potentially we can save three wrong answers and one correct one. We want all that data.

Reply:
I put it in `useEffect` because `isCorrect` is updated in `handleSubmit` and it stores both true and false responses as an object. I did that to handle the multiple questions in the second round of Takeaway questions. A reason why you would see only correct responses is that you cannot progress unless you get the question right. Though you should be able to get more than 2 wrong answers if you answer incorrectly then switch to another dataset.

I've made some corrections according to the other comments, let me know if it works more like you would expect it to.

Reply:
Yeah I think the variable names are a little confusing here. isCorrect is something you would assign to a boolean variable.
And I would still move the definition of asyncSave function outside. As well as have the saving be tied to the user action like the handleSubmit.

Reply:
I am also wondering if it would make sense for a user to hit some sort of continue button before seeing the Takeaways section, something to signal to the user that there are now questions to answer as well as providing a good spot to create the submission... Maybe we can discuss later on if loading that submit button is an issue

Reply:
I can ask Zarina about a `Continue` button. That would be a good place to set up the `submissionId`, but it might mess with her vision for how the simulation flows.

As for the save function: I'll move it outside of the definition out of the function, but I'm worried about race conditions in the `handleSubmit` function. Results need to save after the `isCorrect` object updates and the `submissionId` is created. My plan was to have it trigger in a `useEffect` as a direct response to `handleSubmit`. Alternatively I could run `then()` on a returning `Promise` from the `useState` functions in `handleSubmit`. Do you think we can get away with running all of this in the `handleSubmit` function or do I need to chain this sequence some other way?

I'll rename `isCorrect` to `results` or something.